src/bidiMapper/domains/context/BrowsingContextProcessor.ts

```diff
@@ -434,19 +429,17 @@ export class BrowsingContextProcessor {
   #handleWorkerTarget(cdpTarget: CdpTarget, browsingContextId: string) {
     cdpTarget.cdpClient.on('Runtime.executionContextCreated', (params) => {
       const {uniqueId, id, origin} = params.context;
-      const realm = new Realm(
-        this.#realmStorage,
-        this.#browsingContextStorage,
-        uniqueId,
-        browsingContextId,
-        id,
-        serializeOrigin(origin),
-        'dedicated-worker',
-        undefined,
+      const realm = new DedicatedWorkerRealm(
         cdpTarget.cdpClient,
         this.#eventManager,
-        this.#sharedIdWithFrame,
-        this.#logger
+        id,
+        this.#logger,
+        serializeOrigin(origin),
+        this.#realmStorage.getRealm({
+          browsingContextId,
+        }) as WindowRealm,
```

Comment:
```suggestion
        this.#realmStorage.getRealm({
          browsingContextId,
        }),
```

Reply:
This doesn't help because `getRealm` is not typed.

Reply:
Why `DedicatedWorkerRealm` should care of the owner realm type?

Reply:
This comment is obsolete anyway. We already changed it to be generic in other places.